src/org/zaproxy/zap/extension/ascanrulesAlpha/AbstractAppFilePlugin.java

```diff
@@ -0,0 +1,142 @@
+                if (isStop()) {
+                        if (LOG.isDebugEnabled()) {
+                                LOG.debug("Scanner " + getName() + " Stopping.");
+                        }
+                        return;
+                }
+
+                HttpMessage newRequest = getNewMsg();
+                newRequest.getRequestHeader().setMethod(HttpRequestHeader.GET);
+                URI baseUri = getBaseMsg().getRequestHeader().getURI();
+                URI elmahUri = null;
+                try {
+                        String baseUriPath = baseUri.getPath() == null ? "" : baseUri.getPath();
+                        elmahUri = new URI(baseUri.getScheme(), null, baseUri.getHost(), baseUri.getPort(),
+                                        createTestablePath(baseUriPath));
+                } catch (URIException uEx) {
+                        if (LOG.isDebugEnabled()) {
+                                LOG.debug("An error occurred creating a URI for the: " + getName() + " scanner. " + uEx.getMessage(),
+                                                uEx);
+                        }
+                        return;
+                }
+                try {
+                        newRequest.getRequestHeader().setURI(elmahUri);
+                } catch (URIException uEx) {
+                        if (LOG.isDebugEnabled()) {
+                                LOG.debug("An error occurred setting the URI for a new request used by: " + getName() + " scanner. "
+                                                + uEx.getMessage(), uEx);
+                        }
+			return;
+                }
+                // Until https://github.com/zaproxy/zaproxy/issues/3563 is addressed track completed in Kb
+                // TODO change this when possible
+                synchronized (getKb()) {
+                    if (getKb().getBoolean(elmahUri, MESSAGE_PREFIX)) {
+                        return;
+                    }
+                    getKb().add(elmahUri, MESSAGE_PREFIX, Boolean.TRUE);
+                }
+                try {
+                        sendAndReceive(newRequest, false);
+                } catch (IOException e) {
+                        LOG.warn("An error occurred while checking [" + newRequest.getRequestHeader().getMethod() + "] ["
+                                        + newRequest.getRequestHeader().getURI() + "] for " + getName() + " Caught "
+                                        + e.getClass().getName() + " " + e.getMessage());
+                        return;
+                }
+                int statusCode = newRequest.getResponseHeader().getStatusCode();
+                if (statusCode == HttpStatusCode.OK) {
+                        raiseAlert(newRequest, getRisk(), "");
+                } else if (statusCode == HttpStatusCode.UNAUTHORIZED || statusCode == HttpStatusCode.FORBIDDEN) {
+                        raiseAlert(newRequest, Alert.RISK_INFO, getOtherInfo());
+                }
```

Comment:
Will this status code and risk handling always be relevant/generic?

Reply:
For these two scanners yes, if we happen to need different behaviour we could change later? (This was to avoid duplication between the two scanners.)

Reply:
Yup that works.